business/layer.go

```diff
@@ -165,29 +156,29 @@ func SetWithBackends(cf kubernetes.ClientFactory, prom prometheus.ClientInterfac
 	prometheusClient = prom
 }
 
-// NewWithBackends creates the business layer using the passed k8s and prom clients.
+// NewWithBackends creates the business layer using the passed k8sClients and prom clients.
 // Note that the client passed here should *not* be the Kiali ServiceAccount client.
 // It should be the user client based on the logged in user's token.
-func NewWithBackends(k8s kubernetes.ClientInterface, prom prometheus.ClientInterface, jaegerClient JaegerLoader) *Layer {
+func NewWithBackends(k8s map[string]kubernetes.ClientInterface, kialiSAClients map[string]kubernetes.ClientInterface, prom prometheus.ClientInterface, jaegerClient JaegerLoader) *Layer {
 	temporaryLayer := &Layer{}
-	temporaryLayer.App = AppService{prom: prom, k8s: k8s, businessLayer: temporaryLayer}
+	temporaryLayer.App = AppService{prom: prom, k8s: k8s[kubernetes.HomeClusterName], businessLayer: temporaryLayer}
```

Comment:
maybe its just me not understanding this function, but... why is the home cluster hardcoded in all of these? All the clients will be for the home cluster...should there be a TODO here to update this later for multicluster (I would think we'd need to pass a cluster name to this function)?

Reply:
You are right, we need a TODO here. This was to delimit this task. Not sure if other services will need all the k8s structure, because e.g. for Jaeger, there is going to be just the local service, even though, probably it will need access to all the k8s client for the clusters.

Reply:
Added TODO